File: api/resources/product/product_model.go
package product

import (
	"time"

	"gopkg.in/mgo.v2/bson"
)

// Product represents information about product
type Product struct {
	ID            bson.ObjectId `json:"id" bson:"_id"`
	Name          string        `json:"name"`
	Manufacturer  string        `json:"manufacturer"`
	Ean           string        `json:"ean"`
	Energy        string        `json:"energy"`
	NetWeight     string        `json:"netWeight"`
	UnitOfMeasure string        `json:"unitOfMeasure"`
	Price         Prices        `json:"price"`
}

// Products represents information about product list
type Products []Product

type Price struct {
	Date  time.Time `json:"date"`
	Value float32   `json:"value"`
}

type Prices []Price

File: old_web/src/app/product/product.model.ts
import { Price } from './price.model'

export class Product {
    constructor(
        private id: string,

        public name: string,
        public price: Price[],

        public manufacturer?: string,
        public ean?: string,
        public energy?: string,
        public netWeight?: string,
        public unitOfMeasure?: string
    ) { }

    public getId(): string {
        return this.id;
    }
}